// stepwar-ui/src/App.tsx
"use client";

import React, { useState, useRef, useEffect } from 'react';
import { motion, AnimatePresence } from 'framer-motion';
import { 
  Home, 
  Trophy, 
  Swords, 
  Target, 
  User, 
  Plus, 
  QrCode, 
  Search, 
  MapPin, 
  Users, 
  Crown, 
  Flame, 
  Zap,
  Calendar,
  Award,
  Settings,
  Palette,
  Moon,
  Sun,
  Sparkles
} from 'lucide-react';
import { Button } from './components/ui/button';
import { Input } from './components/ui/input';
import { Card } from './components/ui/card';
import { Badge } from './components/ui/badge';
import { Switch } from './components/ui/switch';
import { Tabs, TabsContent, TabsList, TabsTrigger } from './components/ui/tabs';

// Types
interface User {
  id: string;
  name: string;
  avatar: string;
  level: number;
  xp: number;
  totalSteps: number;
  todaySteps: number;
}

interface League {
  id: string;
  name: string;
  members: number;
  isPublic: boolean;
  leaderboard: { user: User; steps: number }[];
}

interface Battle {
  id: string;
  opponent: User;
  mySteps: number;
  opponentSteps: number;
  status: 'active' | 'won' | 'lost';
  timeLeft: string;
}

interface Challenge {
  id: string;
  title: string;
  type: 'steps' | 'calories' | 'distance' | 'time' | 'battles';
  target: number;
  current: number;
  deadline: string;
}

type Theme = 'black-neon' | 'glassmorphism' | 'light';

// Mock data
const mockUser: User = {
  id: '1',
  name: 'Alex Runner',
  avatar: '👤',
  level: 15,
  xp: 2450,
  totalSteps: 125000,
  todaySteps: 8547
};

const mockWeeklySteps = [
  { day: 'Mon', steps: 8200 },
  { day: 'Tue', steps: 9100 },
  { day: 'Wed', steps: 7800 },
  { day: 'Thu', steps: 10200 },
  { day: 'Fri', steps: 8900 },
  { day: 'Sat', steps: 12000 },
  { day: 'Sun', steps: 8547 }
];

const mockLeagues: League[] = [
  {
    id: '1',
    name: 'City Runners',
    members: 156,
    isPublic: true,
    leaderboard: [
      { user: { ...mockUser, name: 'Sarah Speed' }, steps: 15000 },
      { user: mockUser, steps: 8547 },
      { user: { ...mockUser, name: 'Mike Marathon' }, steps: 7200 }
    ]
  }
];

const mockBattles: Battle[] = [
  {
    id: '1',
    opponent: { ...mockUser, name: 'Jenny Fast' },
    mySteps: 8547,
    opponentSteps: 7200,
    status: 'active',
    timeLeft: '2h 15m'
  }
];

const mockChallenges: Challenge[] = [
  {
    id: '1',
    title: '10K Steps Daily',
    type: 'steps',
    target: 10000,
    current: 8547,
    deadline: '2024-01-31'
  },
  {
    id: '2',
    title: 'Win 5 Battles',
    type: 'battles',
    target: 5,
    current: 2,
    deadline: '2024-02-15'
  }
];

// Theme configurations
const themes = {
  'black-neon': {
    bg: 'bg-black',
    cardBg: 'bg-gray-900/80 backdrop-blur-sm border border-cyan-500/30 shadow-lg shadow-cyan-500/20',
    text: 'text-white',
    accent: 'text-cyan-400',
    button: 'bg-gradient-to-r from-cyan-500 to-blue-500 hover:from-cyan-600 hover:to-blue-600 text-white shadow-lg shadow-cyan-500/30',
    nav: 'bg-gray-900/90 backdrop-blur-md border-t border-cyan-500/30',
    glow: 'shadow-[0_0_20px_rgba(6,182,212,0.3)]'
  },
  'glassmorphism': {
    bg: 'bg-gradient-to-br from-gray-900 via-slate-900 to-black',
    cardBg: 'bg-white/10 backdrop-blur-xl border border-white/20 shadow-2xl shadow-black/20',
    text: 'text-white',
    accent: 'text-blue-200',
    button: 'bg-white/20 hover:bg-white/30 text-white backdrop-blur-xl border border-white/30 shadow-lg shadow-black/10',
    nav: 'bg-black/40 backdrop-blur-xl border-t border-white/20',
    glow: 'shadow-[0_0_30px_rgba(255,255,255,0.1)]'
  },
  'light': {
    bg: 'bg-gradient-to-br from-blue-50 via-purple-50 to-pink-50',
    cardBg: 'bg-white/80 backdrop-blur-sm border border-gray-200 shadow-lg shadow-purple-500/10',
    text: 'text-gray-900',
    accent: 'text-purple-600',
    button: 'bg-gradient-to-r from-purple-500 to-pink-500 hover:from-purple-600 hover:to-pink-600 text-white shadow-lg shadow-purple-500/30',
    nav: 'bg-white/90 backdrop-blur-md border-t border-gray-200',
    glow: 'shadow-[0_0_20px_rgba(147,51,234,0.2)]'
  }
};

// Animated Counter Component
const AnimatedCounter: React.FC<{ value: number; duration?: number }> = ({ value, duration = 2000 }) => {
  const [count, setCount] = useState(0);
  const countRef = useRef(0);

  useEffect(() => {
    const startTime = Date.now();
    const animate = () => {
      const elapsed = Date.now() - startTime;
      const progress = Math.min(elapsed / duration, 1);
      const easeOut = 1 - Math.pow(1 - progress, 3);
      const currentCount = Math.floor(easeOut * value);
      
      if (currentCount !== countRef.current) {
        countRef.current = currentCount;
        setCount(currentCount);
      }
      
      if (progress < 1) {
        requestAnimationFrame(animate);
      }
    };
    animate();
  }, [value, duration]);

  return <span>{count.toLocaleString()}</span>;
};

// Navigation Component
const BottomNav: React.FC<{ activeTab: string; onTabChange: (tab: string) => void; theme: Theme }> = ({ 
  activeTab, 
  onTabChange, 
  theme 
}) => {
  const themeConfig = themes[theme];
  
  const navItems = [
    { id: 'home', icon: Home, label: 'Home' },
    { id: 'leagues', icon: Trophy, label: 'Leagues' },
    { id: 'battles', icon: Swords, label: 'Battles' },
    { id: 'challenges', icon: Target, label: 'Challenges' },
    { id: 'profile', icon: User, label: 'Profile' }
  ];

  return (
    <motion.nav 
      className={`fixed bottom-0 left-0 right-0 ${themeConfig.nav} p-4 z-50`}
      initial={{ y: 100 }}
      animate={{ y: 0 }}
      transition={{ type: "spring", stiffness: 300, damping: 30 }}
    >
      <div className="flex justify-around items-center max-w-md mx-auto">
        {navItems.map((item) => {
          const Icon = item.icon;
          const isActive = activeTab === item.id;
          
          return (
            <motion.button
              key={item.id}
              onClick={() => onTabChange(item.id)}
              className={`relative flex flex-col items-center p-2 rounded-xl transition-all ${
                isActive ? themeConfig.accent : themeConfig.text + ' opacity-60'
              }`}
              whileHover={{ scale: 1.1 }}
              whileTap={{ scale: 0.95 }}
            >
              {isActive && (
                <motion.div
                  layoutId="activeTab"
                  className={`absolute inset-0 rounded-xl ${
                    theme === 'black-neon' ? 'bg-cyan-500/20' :
                    theme === 'glassmorphism' ? 'bg-white/20' :
                    'bg-purple-500/20'
                  }`}
                  transition={{ type: "spring", stiffness: 300, damping: 30 }}
                />
              )}
              <Icon size={24} className={`relative z-10 ${isActive && themeConfig.glow}`} />
              <span className="text-xs mt-1 relative z-10">{item.label}</span>
            </motion.button>
          );
        })}
      </div>
    </motion.nav>
  );
};

// Home Screen Component
const HomeScreen: React.FC<{ theme: Theme; user: User; onAddSteps: (steps: number) => void }> = ({ theme, user, onAddSteps }) => {
  const themeConfig = themes[theme];
  
  return (
    <div className="space-y-6 pb-24">
      {/* Today's Steps */}
      <motion.div
        initial={{ opacity: 0, y: 20 }}
        animate={{ opacity: 1, y: 0 }}
        className={`${themeConfig.cardBg} rounded-3xl p-8 text-center ${themeConfig.glow}`}
      >
        <h2 className={`text-lg font-medium ${themeConfig.text} mb-4`}>Today's Steps</h2>
        <div className={`text-6xl font-bold ${themeConfig.accent} mb-2`}>
          <AnimatedCounter value={user.todaySteps} />
        </div>
        <div className={`${themeConfig.text} opacity-70`}>
          Goal: 10,000 steps
        </div>
        <div className="w-full bg-gray-700 rounded-full h-2 mt-4">
          <motion.div
            className={`h-2 rounded-full ${
              theme === 'black-neon' ? 'bg-gradient-to-r from-cyan-400 to-blue-500' :
              theme === 'glassmorphism' ? 'bg-gradient-to-r from-purple-400 to-pink-400' :
              'bg-gradient-to-r from-purple-500 to-pink-500'
            }`}
            initial={{ width: 0 }}
            animate={{ width: `${Math.min((user.todaySteps / 10000) * 100, 100)}%` }}
            transition={{ duration: 1.5, ease: "easeOut" }}
          />
        </div>
        
        {/* Add Steps Button */}
        <motion.div className="mt-6">
          <Button 
            onClick={() => onAddSteps(100)}
            className={`${themeConfig.button} w-full`}
            whileHover={{ scale: 1.05 }}
            whileTap={{ scale: 0.95 }}
          >
            <Zap className="w-4 h-4 mr-2" />
            Add 100 Steps
          </Button>
        </motion.div>
      </motion.div>

      {/* Weekly Steps */}
      <motion.div
        initial={{ opacity: 0, y: 20 }}
        animate={{ opacity: 1, y: 0 }}
        transition={{ delay: 0.1 }}
        className={`${themeConfig.cardBg} rounded-3xl p-6`}
      >
        <h3 className={`text-lg font-semibold ${themeConfig.text} mb-4`}>This Week</h3>
        <div className="flex justify-between items-end h-32">
          {mockWeeklySteps.map((day, index) => (
            <div key={day.day} className="flex flex-col items-center">
              <motion.div
                className={`w-8 rounded-t-lg ${
                  theme === 'black-neon' ? 'bg-gradient-to-t from-cyan-500 to-blue-400' :
                  theme === 'glassmorphism' ? 'bg-gradient-to-t from-purple-400 to-pink-400' :
                  'bg-gradient-to-t from-purple-500 to-pink-500'
                }`}
                initial={{ height: 0 }}
                animate={{ height: `${(day.steps / 12000) * 100}px` }}
                transition={{ delay: 0.2 + index * 0.1, duration: 0.8 }}
              />
              <span className={`text-xs ${themeConfig.text} opacity-70 mt-2`}>{day.day}</span>
            </div>
          ))}
        </div>
      </motion.div>

      {/* Personal Best */}
      <motion.div
        initial={{ opacity: 0, y: 20 }}
        animate={{ opacity: 1, y: 0 }}
        transition={{ delay: 0.2 }}
        className={`${themeConfig.cardBg} rounded-3xl p-6`}
      >
        <div className="flex items-center justify-between">
          <div>
            <h3 className={`text-lg font-semibold ${themeConfig.text}`}>Personal Best</h3>
            <p className={`text-2xl font-bold ${themeConfig.accent}`}>15,247 steps</p>
            <p className={`text-sm ${themeConfig.text} opacity-70`}>December 15, 2023</p>
          </div>
          <Award className={`w-12 h-12 ${themeConfig.accent}`} />
        </div>
      </motion.div>

      {/* Quick Access */}
      <motion.div
        initial={{ opacity: 0, y: 20 }}
        animate={{ opacity: 1, y: 0 }}
        transition={{ delay: 0.3 }}
        className="grid grid-cols-3 gap-4"
      >
        <div className={`${themeConfig.cardBg} rounded-2xl p-4 text-center`}>
          <Trophy className={`w-8 h-8 ${themeConfig.accent} mx-auto mb-2`} />
          <p className={`text-sm ${themeConfig.text}`}>Leagues</p>
          <p className={`text-xs ${themeConfig.text} opacity-70`}>3 active</p>
        </div>
        <div className={`${themeConfig.cardBg} rounded-2xl p-4 text-center`}>
          <Swords className={`w-8 h-8 ${themeConfig.accent} mx-auto mb-2`} />
          <p className={`text-sm ${themeConfig.text}`}>Battles</p>
          <p className={`text-xs ${themeConfig.text} opacity-70`}>2 ongoing</p>
        </div>
        <div className={`${themeConfig.cardBg} rounded-2xl p-4 text-center`}>
          <Target className={`w-8 h-8 ${themeConfig.accent} mx-auto mb-2`} />
          <p className={`text-sm ${themeConfig.text}`}>Challenges</p>
          <p className={`text-xs ${themeConfig.text} opacity-70`}>5 active</p>
        </div>
      </motion.div>
    </div>
  );
};

// Leagues Screen Component
const LeaguesScreen: React.FC<{ 
  theme: Theme; 
  leagues: League[]; 
  user: User; 
  showCreateModal: boolean; 
  showJoinModal: boolean; 
  onShowCreateModal: (show: boolean) => void; 
  onShowJoinModal: (show: boolean) => void; 
  newLeagueName: string; 
  newLeagueIsPublic: boolean; 
  onNewLeagueNameChange: (name: string) => void; 
  onNewLeagueIsPublicChange: (isPublic: boolean) => void; 
  onCreateLeague: () => void;
  searchQuery: string;
  searchResults: League[];
  onSearch: (query: string) => void;
  joinCode: string;
  onJoinCodeChange: (code: string) => void;
  onJoinByCode: () => void;
  nearbyLeagues: League[];
  onFindNearby: () => void;
  onLeagueClick: (league: League) => void;
}> = ({ 
  theme, 
  leagues, 
  user, 
  showCreateModal, 
  showJoinModal, 
  onShowCreateModal, 
  onShowJoinModal, 
  newLeagueName, 
  newLeagueIsPublic, 
  onNewLeagueNameChange, 
  onNewLeagueIsPublicChange, 
  onCreateLeague,
  searchQuery,
  searchResults,
  onSearch,
  joinCode,
  onJoinCodeChange,
  onJoinByCode,
  nearbyLeagues,
  onFindNearby,
  onLeagueClick
}) => {
  const themeConfig = themes[theme];

  return (
    <div className="space-y-6 pb-24">
      {/* Header */}
      <div className="flex justify-between items-center">
        <h1 className={`text-2xl font-bold ${themeConfig.text}`}>Leagues</h1>
        <div className="flex gap-2">
          <Button 
            onClick={() => onShowJoinModal(true)}
            variant="outline" 
            size="sm"
            className={themeConfig.button}
          >
            <Search className="w-4 h-4 mr-2" />
            Join
          </Button>
          <Button 
            onClick={() => onShowCreateModal(true)}
            size="sm"
            className={themeConfig.button}
          >
            <Plus className="w-4 h-4 mr-2" />
            Create
          </Button>
        </div>
      </div>

      {/* My Leagues */}
      <div className="space-y-4">
        {leagues.map((league) => (
          <motion.div
            key={league.id}
            initial={{ opacity: 0, x: -20 }}
            animate={{ opacity: 1, x: 0 }}
            className={`${themeConfig.cardBg} rounded-3xl p-6`}
            onClick={() => onLeagueClick(league)}
          >
            <div className="flex justify-between items-start mb-4">
              <div>
                <h3 className={`text-lg font-semibold ${themeConfig.text}`}>{league.name}</h3>
                <p className={`text-sm ${themeConfig.text} opacity-70`}>
                  {league.members} members • {league.isPublic ? 'Public' : 'Private'}
                </p>
              </div>
              <Crown className={`w-6 h-6 ${themeConfig.accent}`} />
            </div>
            
            <div className="space-y-3">
              <h4 className={`text-sm font-medium ${themeConfig.text}`}>Leaderboard</h4>
              {league.leaderboard.map((entry, index) => (
                <div key={index} className="flex items-center justify-between">
                  <div className="flex items-center gap-3">
                    <span className={`text-lg ${themeConfig.accent}`}>#{index + 1}</span>
                    <span className={`text-2xl`}>{entry.user.avatar}</span>
                    <span className={`${themeConfig.text}`}>{entry.user.name}</span>
                  </div>
                  <span className={`font-semibold ${themeConfig.accent}`}>
                    {entry.steps.toLocaleString()}
                  </span>
                </div>
              ))}
            </div>
          </motion.div>
        ))}
      </div>

      {/* Create League Modal */}
      <AnimatePresence>
        {showCreateModal && (
          <motion.div
            initial={{ opacity: 0 }}
            animate={{ opacity: 1 }}
            exit={{ opacity: 0 }}
            className="fixed inset-0 bg-black/50 backdrop-blur-sm z-50 flex items-center justify-center p-4"
            onClick={() => onShowCreateModal(false)}
          >
            <motion.div
              initial={{ scale: 0.9, opacity: 0 }}
              animate={{ scale: 1, opacity: 1 }}
              exit={{ scale: 0.9, opacity: 0 }}
              className={`${themeConfig.cardBg} rounded-3xl p-6 w-full max-w-md`}
              onClick={(e) => e.stopPropagation()}
            >
              <h2 className={`text-xl font-bold ${themeConfig.text} mb-4`}>Create League</h2>
              <div className="space-y-4">
                <Input placeholder="League Name" className={themeConfig.text} value={newLeagueName} onChange={(e) => onNewLeagueNameChange(e.target.value)} />
                <div className="flex items-center justify-between">
                  <span className={themeConfig.text}>Public League</span>
                  <Switch checked={newLeagueIsPublic} onCheckedChange={onNewLeagueIsPublicChange} />
                </div>
                <div className="flex gap-2">
                  <Button 
                    variant="outline" 
                    onClick={() => onShowCreateModal(false)}
                    className="flex-1"
                  >
                    Cancel
                  </Button>
                  <Button className={`flex-1 ${themeConfig.button}`} onClick={onCreateLeague}>
                    Create
                  </Button>
                </div>
              </div>
            </motion.div>
          </motion.div>
        )}
      </AnimatePresence>

      {/* Join League Modal */}
      <AnimatePresence>
        {showJoinModal && (
          <motion.div
            initial={{ opacity: 0 }}
            animate={{ opacity: 1 }}
            exit={{ opacity: 0 }}
            className="fixed inset-0 bg-black/50 backdrop-blur-sm z-50 flex items-center justify-center p-4"
            onClick={() => onShowJoinModal(false)}
          >
            <motion.div
              initial={{ scale: 0.9, opacity: 0 }}
              animate={{ scale: 1, opacity: 1 }}
              exit={{ scale: 0.9, opacity: 0 }}
              className={`${themeConfig.cardBg} rounded-3xl p-6 w-full max-w-md`}
              onClick={(e) => e.stopPropagation()}
            >
              <h2 className={`text-xl font-bold ${themeConfig.text} mb-4`}>Join League</h2>
              <Tabs defaultValue="search" className="w-full">
                <TabsList className="grid w-full grid-cols-3">
                  <TabsTrigger value="search">Search</TabsTrigger>
                  <TabsTrigger value="location">Location</TabsTrigger>
                  <TabsTrigger value="code">Code</TabsTrigger>
                </TabsList>
                <TabsContent value="search" className="space-y-4">
                  <Input 
                    placeholder="Search leagues..." 
                    value={searchQuery}
                    onChange={(e) => onSearch(e.target.value)}
                  />
                  {searchResults.length > 0 && (
                    <div className="space-y-2 max-h-40 overflow-y-auto">
                      {searchResults.map((league) => (
                        <div key={league.id} className={`p-3 rounded-lg bg-white/10 border border-white/20 ${themeConfig.text}`}>
                          <div className="font-medium">{league.name}</div>
                          <div className="text-sm opacity-70">{league.members} members</div>
                        </div>
                      ))}
                    </div>
                  )}
                  <Button className={`w-full ${themeConfig.button}`} onClick={() => onSearch(searchQuery)}>
                    Search
                  </Button>
                </TabsContent>
                <TabsContent value="location" className="space-y-4">
                  <div className="flex items-center gap-2">
                    <MapPin className={`w-4 h-4 ${themeConfig.accent}`} />
                    <span className={themeConfig.text}>Find leagues near you</span>
                  </div>
                  {nearbyLeagues.length > 0 && (
                    <div className="space-y-2 max-h-40 overflow-y-auto">
                      {nearbyLeagues.map((league) => (
                        <div key={league.id} className={`p-3 rounded-lg bg-white/10 border border-white/20 ${themeConfig.text}`}>
                          <div className="font-medium">{league.name}</div>
                          <div className="text-sm opacity-70">{league.members} members</div>
                        </div>
                      ))}
                    </div>
                  )}
                  <Button className={`w-full ${themeConfig.button}`} onClick={onFindNearby}>
                    Find Nearby
                  </Button>
                </TabsContent>
                <TabsContent value="code" className="space-y-4">
                  <Input 
                    placeholder="Enter league code..." 
                    value={joinCode}
                    onChange={(e) => onJoinCodeChange(e.target.value)}
                  />
                  <Button className={`w-full ${themeConfig.button}`} onClick={onJoinByCode}>
                    Join
                  </Button>
                </TabsContent>
              </Tabs>
            </motion.div>
          </motion.div>
        )}
      </AnimatePresence>
    </div>
  );
};

// Battles Screen Component
const BattlesScreen: React.FC<{ 
  theme: Theme; 
  battles: Battle[]; 
  user: User; 
  showCreateModal: boolean; 
  onShowCreateModal: (show: boolean) => void; 
  onCreateBattle: (opponentName: string, duration: '24h' | '7d') => void;
  newOpponent: string;
  newDuration: '24h' | '7d';
  onNewOpponentChange: (name: string) => void;
  onNewDurationChange: (duration: '24h' | '7d') => void;
  onBattleClick: (battle: Battle) => void;
}> = ({ 
  theme, 
  battles, 
  user, 
  showCreateModal, 
  onShowCreateModal, 
  onCreateBattle, 
  newOpponent,
  newDuration,
  onNewOpponentChange,
  onNewDurationChange,
  onBattleClick
}) => {
  const themeConfig = themes[theme];

  // Fiery theme override for battles
  const battleTheme = {
    ...themeConfig,
    cardBg: theme === 'black-neon' 
      ? 'bg-gradient-to-br from-red-900/80 to-orange-900/80 backdrop-blur-sm border border-red-500/30 shadow-lg shadow-red-500/20'
      : theme === 'glassmorphism'
      ? 'bg-gradient-to-br from-red-500/20 to-orange-500/20 backdrop-blur-xl border border-red-300/30 shadow-2xl shadow-black/20'
      : 'bg-gradient-to-br from-red-100 to-orange-100 backdrop-blur-sm border border-red-200 shadow-lg shadow-red-500/10',
    accent: theme === 'light' ? 'text-red-600' : 'text-red-400',
    glow: 'shadow-[0_0_20px_rgba(239,68,68,0.3)]'
  };

  return (
    <div className="space-y-6 pb-24">
      {/* Header with fiery styling */}
      <div className="flex justify-between items-center">
        <div className="flex items-center gap-2">
          <Flame className={`w-8 h-8 ${battleTheme.accent}`} />
          <h1 className={`text-2xl font-bold ${themeConfig.text}`}>Battles</h1>
        </div>
        <div className="flex gap-2">
          <Button 
            size="sm"
            className="bg-gradient-to-r from-red-500 to-orange-500 hover:from-red-600 hover:to-orange-600 text-white"
          >
            <QrCode className="w-4 h-4 mr-2" />
            Scan QR
          </Button>
          <Button 
            onClick={() => onShowCreateModal(true)}
            size="sm"
            className="bg-gradient-to-r from-orange-500 to-red-500 hover:from-orange-600 hover:to-red-600 text-white"
          >
            <Plus className="w-4 h-4 mr-2" />
            Create
          </Button>
        </div>
      </div>

      {/* Active Battles */}
      <div className="space-y-4">
        {battles.map((battle) => (
          <motion.div
            key={battle.id}
            initial={{ opacity: 0, scale: 0.95 }}
            animate={{ opacity: 1, scale: 1 }}
            className={`${battleTheme.cardBg} rounded-3xl p-6 ${battleTheme.glow}`}
            onClick={() => onBattleClick(battle)}
          >
            <div className="flex items-center justify-between mb-4">
              <div className="flex items-center gap-3">
                <span className="text-2xl">👤</span>
                <div>
                  <h3 className={`font-semibold ${themeConfig.text}`}>{battle.opponent.name}</h3>
                  <p className={`text-sm ${themeConfig.text} opacity-70`}>1v1 Battle</p>
                </div>
              </div>
              <div className="text-right">
                <p className={`text-sm ${battleTheme.accent} font-medium`}>Time Left</p>
                <p className={`text-lg font-bold ${battleTheme.accent}`}>{battle.timeLeft}</p>
              </div>
            </div>

            <div className="space-y-3">
              <div className="flex justify-between items-center">
                <span className={themeConfig.text}>You</span>
                <span className={`font-bold ${battleTheme.accent}`}>
                  {battle.mySteps.toLocaleString()} steps
                </span>
              </div>
              <div className="flex justify-between items-center">
                <span className={themeConfig.text}>{battle.opponent.name}</span>
                <span className={`font-bold ${themeConfig.text} opacity-70`}>
                  {battle.opponentSteps.toLocaleString()} steps
                </span>
              </div>
              
              {/* Battle progress bar */}
              <div className="relative h-3 bg-gray-700 rounded-full overflow-hidden">
                <motion.div
                  className="absolute left-0 top-0 h-full bg-gradient-to-r from-red-500 to-orange-500"
                  initial={{ width: 0 }}
                  animate={{ 
                    width: `${(battle.mySteps / (battle.mySteps + battle.opponentSteps)) * 100}%` 
                  }}
                  transition={{ duration: 1.5, ease: "easeOut" }}
                />
              </div>
              
              <div className="flex items-center justify-center gap-2">
                {battle.mySteps > battle.opponentSteps ? (
                  <>
                    <Zap className={`w-4 h-4 ${battleTheme.accent}`} />
                    <span className={`text-sm font-medium ${battleTheme.accent}`}>
                      You're winning!
                    </span>
                  </>
                ) : (
                  <span className={`text-sm ${themeConfig.text} opacity-70`}>
                    Keep pushing!
                  </span>
                )}
              </div>
            </div>
          </motion.div>
        ))}
      </div>

      {/* Create Battle Modal */}
      <AnimatePresence>
        {showCreateModal && (
          <motion.div
            initial={{ opacity: 0 }}
            animate={{ opacity: 1 }}
            exit={{ opacity: 0 }}
            className="fixed inset-0 bg-black/50 backdrop-blur-sm z-50 flex items-center justify-center p-4"
            onClick={() => onShowCreateModal(false)}
          >
            <motion.div
              initial={{ scale: 0.9, opacity: 0 }}
              animate={{ scale: 1, opacity: 1 }}
              exit={{ scale: 0.9, opacity: 0 }}
              className={`${battleTheme.cardBg} rounded-3xl p-6 w-full max-w-md`}
              onClick={(e) => e.stopPropagation()}
            >
              <div className="flex items-center gap-2 mb-4">
                <Flame className={`w-6 h-6 ${battleTheme.accent}`} />
                <h2 className={`text-xl font-bold ${themeConfig.text}`}>Create Battle</h2>
              </div>
              <div className="space-y-4">
                <Input 
                  placeholder="Challenge a friend..." 
                  className={themeConfig.text}
                  value={newOpponent}
                  onChange={(e) => onNewOpponentChange(e.target.value)}
                />
                <div className="grid grid-cols-2 gap-2">
                  <Button 
                    variant="outline" 
                    onClick={() => onCreateBattle(newOpponent || 'Friend', '24h')}
                    className={`flex-1 ${newDuration === '24h' ? 'bg-white/20' : ''}`}
                  >
                    24 Hours
                  </Button>
                  <Button 
                    variant="outline" 
                    onClick={() => onCreateBattle(newOpponent || 'Friend', '7d')}
                    className={`flex-1 ${newDuration === '7d' ? 'bg-white/20' : ''}`}
                  >
                    7 Days
                  </Button>
                </div>
                <div className="flex gap-2">
                  <Button 
                    variant="outline" 
                    onClick={() => onShowCreateModal(false)}
                    className="flex-1"
                  >
                    Cancel
                  </Button>
                  <Button 
                    className="flex-1 bg-gradient-to-r from-red-500 to-orange-500 hover:from-red-600 hover:to-orange-600 text-white"
                    onClick={() => onCreateBattle(newOpponent || 'Random Opponent', newDuration)}
                  >
                    Challenge!
                  </Button>
                </div>
              </div>
            </motion.div>
          </motion.div>
        )}
      </AnimatePresence>
    </div>
  );
};

// Challenges Screen Component
const ChallengesScreen: React.FC<{ 
  theme: Theme; 
  challenges: Challenge[]; 
  showCreateModal: boolean; 
  onShowCreateModal: (show: boolean) => void; 
  newChallengeTitle: string; 
  newChallengeType: 'steps' | 'calories' | 'distance' | 'time' | 'battles'; 
  newChallengeTarget: number; 
  newChallengeDeadline: string; 
  onNewChallengeTitleChange: (title: string) => void; 
  onNewChallengeTypeChange: (type: 'steps' | 'calories' | 'distance' | 'time' | 'battles') => void; 
  onNewChallengeTargetChange: (target: number) => void; 
  onNewChallengeDeadlineChange: (deadline: string) => void; 
  onCreateChallenge: () => void;
  onUpdateProgress: (challengeId: string, progress: number) => void;
  onChallengeClick: (challenge: Challenge) => void;
}> = ({ 
  theme, 
  challenges, 
  showCreateModal, 
  onShowCreateModal, 
  newChallengeTitle, 
  newChallengeType, 
  newChallengeTarget, 
  newChallengeDeadline, 
  onNewChallengeTitleChange, 
  onNewChallengeTypeChange, 
  onNewChallengeTargetChange, 
  onNewChallengeDeadlineChange, 
  onCreateChallenge, 
  onUpdateProgress,
  onChallengeClick
}) => {
  const themeConfig = themes[theme];

  return (
    <div className="space-y-6 pb-24">
      <div className="flex justify-between items-center">
        <h1 className={`text-2xl font-bold ${themeConfig.text}`}>Challenges</h1>
        <Button size="sm" className={themeConfig.button}>
          <Plus className="w-4 h-4 mr-2" />
          Create
        </Button>
      </div>

      <div className="space-y-4">
        {challenges.map((challenge) => (
          <motion.div
            key={challenge.id}
            initial={{ opacity: 0, y: 20 }}
            animate={{ opacity: 1, y: 0 }}
            className={`${themeConfig.cardBg} rounded-3xl p-6`}
            onClick={() => onChallengeClick(challenge)}
          >
            <div className="flex justify-between items-start mb-4">
              <div>
                <h3 className={`text-lg font-semibold ${themeConfig.text}`}>{challenge.title}</h3>
                <p className={`text-sm ${themeConfig.text} opacity-70`}>
                  Due: {new Date(challenge.deadline).toLocaleDateString()}
                </p>
              </div>
              <Badge variant="outline" className={themeConfig.accent}>
                {challenge.type}
              </Badge>
            </div>
            
            <div className="space-y-2">
              <div className="flex justify-between">
                <span className={`text-sm ${themeConfig.text}`}>Progress</span>
                <span className={`text-sm ${themeConfig.accent}`}>
                  {challenge.current} / {challenge.target}
                </span>
              </div>
              <div className="w-full bg-gray-700 rounded-full h-2">
                <motion.div
                  className={`h-2 rounded-full ${
                    theme === 'black-neon' ? 'bg-gradient-to-r from-cyan-400 to-blue-500' :
                    theme === 'glassmorphism' ? 'bg-gradient-to-r from-purple-400 to-pink-400' :
                    'bg-gradient-to-r from-purple-500 to-pink-500'
                  }`}
                  initial={{ width: 0 }}
                  animate={{ width: `${(challenge.current / challenge.target) * 100}%` }}
                  transition={{ duration: 1.5, ease: "easeOut" }}
                />
              </div>
              
              {/* Progress Update Buttons */}
              <div className="flex gap-2 mt-4">
                <Button 
                  size="sm"
                  variant="outline"
                  onClick={() => onUpdateProgress(challenge.id, 100)}
                  className="flex-1"
                >
                  +100
                </Button>
                <Button 
                  size="sm"
                  variant="outline"
                  onClick={() => onUpdateProgress(challenge.id, 500)}
                  className="flex-1"
                >
                  +500
                </Button>
                <Button 
                  size="sm"
                  variant="outline"
                  onClick={() => onUpdateProgress(challenge.id, 1000)}
                  className="flex-1"
                >
                  +1000
                </Button>
              </div>
            </div>
          </motion.div>
        ))}
      </div>

      {/* Create Challenge Modal */}
      <AnimatePresence>
        {showCreateModal && (
          <motion.div
            initial={{ opacity: 0 }}
            animate={{ opacity: 1 }}
            exit={{ opacity: 0 }}
            className="fixed inset-0 bg-black/50 backdrop-blur-sm z-50 flex items-center justify-center p-4"
            onClick={() => onShowCreateModal(false)}
          >
            <motion.div
              initial={{ scale: 0.9, opacity: 0 }}
              animate={{ scale: 1, opacity: 1 }}
              exit={{ scale: 0.9, opacity: 0 }}
              className={`${themeConfig.cardBg} rounded-3xl p-6 w-full max-w-md`}
              onClick={(e) => e.stopPropagation()}
            >
              <h2 className={`text-xl font-bold ${themeConfig.text} mb-4`}>Create Challenge</h2>
              <div className="space-y-4">
                <Input 
                  placeholder="Challenge title..." 
                  className={themeConfig.text} 
                  value={newChallengeTitle}
                  onChange={(e) => onNewChallengeTitleChange(e.target.value)}
                />
                
                <div className="space-y-2">
                  <label className={`text-sm ${themeConfig.text}`}>Challenge Type</label>
                  <select 
                    className={`w-full p-2 rounded-lg bg-white/10 border border-white/20 text-white ${themeConfig.text}`}
                    value={newChallengeType}
                    onChange={(e) => onNewChallengeTypeChange(e.target.value as any)}
                  >
                    <option value="steps">Steps</option>
                    <option value="calories">Calories</option>
                    <option value="distance">Distance</option>
                    <option value="time">Time</option>
                    <option value="battles">Battles</option>
                  </select>
                </div>
                
                <div className="space-y-2">
                  <label className={`text-sm ${themeConfig.text}`}>Target</label>
                  <Input 
                    type="number"
                    placeholder="10000" 
                    className={themeConfig.text} 
                    value={newChallengeTarget}
                    onChange={(e) => onNewChallengeTargetChange(parseInt(e.target.value) || 0)}
                  />
                </div>
                
                <div className="space-y-2">
                  <label className={`text-sm ${themeConfig.text}`}>Deadline</label>
                  <Input 
                    type="date"
                    className={themeConfig.text} 
                    value={newChallengeDeadline}
                    onChange={(e) => onNewChallengeDeadlineChange(e.target.value)}
                  />
                </div>
                
                <div className="flex gap-2">
                  <Button 
                    variant="outline" 
                    onClick={() => onShowCreateModal(false)}
                    className="flex-1"
                  >
                    Cancel
                  </Button>
                  <Button 
                    className={`flex-1 ${themeConfig.button}`}
                    onClick={onCreateChallenge}
                  >
                    Create
                  </Button>
                </div>
              </div>
            </motion.div>
          </motion.div>
        )}
      </AnimatePresence>
    </div>
  );
};

// Profile Screen Component
const ProfileScreen: React.FC<{ theme: Theme; onThemeChange: (theme: Theme) => void; user: User; notificationsEnabled: boolean; autoSyncEnabled: boolean; privacyModeEnabled: boolean; onNotificationsChange: (enabled: boolean) => void; onAutoSyncChange: (enabled: boolean) => void; onPrivacyModeChange: (enabled: boolean) => void; }> = ({ 
  theme, 
  onThemeChange, 
  user, 
  notificationsEnabled,
  autoSyncEnabled,
  privacyModeEnabled,
  onNotificationsChange,
  onAutoSyncChange,
  onPrivacyModeChange
}) => {
  const themeConfig = themes[theme];

  return (
    <div className="space-y-6 pb-24">
      <h1 className={`text-2xl font-bold ${themeConfig.text}`}>Profile</h1>

      {/* User Info */}
      <motion.div
        initial={{ opacity: 0, y: 20 }}
        animate={{ opacity: 1, y: 0 }}
        className={`${themeConfig.cardBg} rounded-3xl p-6 text-center`}
      >
        <div className="text-6xl mb-4">{user.avatar}</div>
        <h2 className={`text-xl font-bold ${themeConfig.text}`}>{user.name}</h2>
        <p className={`${themeConfig.accent} font-medium`}>Level {user.level}</p>
        
        <div className="mt-4 space-y-2">
          <div className="flex justify-between">
            <span className={`text-sm ${themeConfig.text}`}>XP Progress</span>
            <span className={`text-sm ${themeConfig.accent}`}>
              {user.xp} / {(user.level + 1) * 1000}
            </span>
          </div>
          <div className="w-full bg-gray-700 rounded-full h-2">
            <motion.div
              className={`h-2 rounded-full ${
                theme === 'black-neon' ? 'bg-gradient-to-r from-cyan-400 to-blue-500' :
                theme === 'glassmorphism' ? 'bg-gradient-to-r from-purple-400 to-pink-400' :
                'bg-gradient-to-r from-purple-500 to-pink-500'
              }`}
              initial={{ width: 0 }}
              animate={{ width: `${(user.xp / ((user.level + 1) * 1000)) * 100}%` }}
              transition={{ duration: 1.5, ease: "easeOut" }}
            />
          </div>
        </div>
      </motion.div>

      {/* Stats */}
      <motion.div
        initial={{ opacity: 0, y: 20 }}
        animate={{ opacity: 1, y: 0 }}
        transition={{ delay: 0.1 }}
        className={`${themeConfig.cardBg} rounded-3xl p-6`}
      >
        <h3 className={`text-lg font-semibold ${themeConfig.text} mb-4`}>Statistics</h3>
        <div className="grid grid-cols-2 gap-4">
          <div className="text-center">
            <p className={`text-2xl font-bold ${themeConfig.accent}`}>
              {user.totalSteps.toLocaleString()}
            </p>
            <p className={`text-sm ${themeConfig.text} opacity-70`}>Total Steps</p>
          </div>
          <div className="text-center">
            <p className={`text-2xl font-bold ${themeConfig.accent}`}>47</p>
            <p className={`text-sm ${themeConfig.text} opacity-70`}>Days Active</p>
          </div>
        </div>
      </motion.div>

      {/* Theme Selector */}
      <motion.div
        initial={{ opacity: 0, y: 20 }}
        animate={{ opacity: 1, y: 0 }}
        transition={{ delay: 0.2 }}
        className={`${themeConfig.cardBg} rounded-3xl p-6`}
      >
        <div className="flex items-center gap-2 mb-4">
          <Palette className={`w-5 h-5 ${themeConfig.accent}`} />
          <h3 className={`text-lg font-semibold ${themeConfig.text}`}>Theme</h3>
        </div>
        
        <div className="space-y-3">
          <div 
            className={`flex items-center justify-between p-3 rounded-xl cursor-pointer transition-all ${
              theme === 'black-neon' ? 'bg-cyan-500/20 border border-cyan-500/30' : 'hover:bg-white/10'
            }`}
            onClick={() => onThemeChange('black-neon')}
          >
            <div className="flex items-center gap-3">
              <div className="w-6 h-6 bg-gradient-to-r from-cyan-400 to-blue-500 rounded-full shadow-lg shadow-cyan-500/30" />
              <div>
                <p className={`font-medium ${themeConfig.text}`}>Black Neon</p>
                <p className={`text-xs ${themeConfig.text} opacity-70`}>Dark with neon highlights</p>
              </div>
            </div>
            {theme === 'black-neon' && <Sparkles className={`w-5 h-5 ${themeConfig.accent}`} />}
          </div>

          <div 
            className={`flex items-center justify-between p-3 rounded-xl cursor-pointer transition-all ${
              theme === 'glassmorphism' ? 'bg-purple-500/20 border border-purple-500/30' : 'hover:bg-white/10'
            }`}
            onClick={() => onThemeChange('glassmorphism')}
          >
            <div className="flex items-center gap-3">
              <div className="w-6 h-6 bg-gradient-to-r from-purple-400 to-pink-400 rounded-full backdrop-blur-sm" />
              <div>
                <p className={`font-medium ${themeConfig.text}`}>Glassmorphism</p>
                <p className={`text-xs ${themeConfig.text} opacity-70`}>Frosted glass effect</p>
              </div>
            </div>
            {theme === 'glassmorphism' && <Sparkles className={`w-5 h-5 ${themeConfig.accent}`} />}
          </div>

          <div 
            className={`flex items-center justify-between p-3 rounded-xl cursor-pointer transition-all ${
              theme === 'light' ? 'bg-purple-500/20 border border-purple-500/30' : 'hover:bg-white/10'
            }`}
            onClick={() => onThemeChange('light')}
          >
            <div className="flex items-center gap-3">
              <div className="w-6 h-6 bg-gradient-to-r from-purple-500 to-pink-500 rounded-full" />
              <div>
                <p className={`font-medium ${themeConfig.text}`}>Light Mode</p>
                <p className={`text-xs ${themeConfig.text} opacity-70`}>Bright with aurora colors</p>
              </div>
            </div>
            {theme === 'light' && <Sparkles className={`w-5 h-5 ${themeConfig.accent}`} />}
          </div>
        </div>
      </motion.div>

      {/* Settings */}
      <motion.div
        initial={{ opacity: 0, y: 20 }}
        animate={{ opacity: 1, y: 0 }}
        transition={{ delay: 0.3 }}
        className={`${themeConfig.cardBg} rounded-3xl p-6`}
      >
        <div className="flex items-center gap-2 mb-4">
          <Settings className={`w-5 h-5 ${themeConfig.accent}`} />
          <h3 className={`text-lg font-semibold ${themeConfig.text}`}>Settings</h3>
        </div>
        
        <div className="space-y-4">
          <div className="flex items-center justify-between">
            <span className={themeConfig.text}>Notifications</span>
            <Switch checked={notificationsEnabled} onCheckedChange={onNotificationsChange} />
          </div>
          <div className="flex items-center justify-between">
            <span className={themeConfig.text}>Auto-sync</span>
            <Switch checked={autoSyncEnabled} onCheckedChange={onAutoSyncChange} />
          </div>
          <div className="flex items-center justify-between">
            <span className={themeConfig.text}>Privacy Mode</span>
            <Switch checked={privacyModeEnabled} onCheckedChange={onPrivacyModeChange} />
          </div>
        </div>
      </motion.div>
    </div>
  );
};

// League Detail Screen Component
const LeagueDetailScreen: React.FC<{ 
  theme: Theme; 
  league: League; 
  user: User;
  onGoBack: () => void;
}> = ({ theme, league, user, onGoBack }) => {
  const themeConfig = themes[theme];

  return (
    <div className="space-y-6 pb-24">
      {/* Header */}
      <div className="flex items-center gap-4">
        <Button 
          onClick={onGoBack}
          variant="ghost"
          size="sm"
          className={themeConfig.text}
        >
          ← Back
        </Button>
        <h1 className={`text-2xl font-bold ${themeConfig.text}`}>{league.name}</h1>
      </div>

      {/* League Info */}
      <motion.div
        initial={{ opacity: 0, y: 20 }}
        animate={{ opacity: 1, y: 0 }}
        className={`${themeConfig.cardBg} rounded-3xl p-6`}
      >
        <div className="flex justify-between items-center mb-4">
          <div>
            <h3 className={`text-lg font-semibold ${themeConfig.text}`}>{league.name}</h3>
            <p className={`text-sm ${themeConfig.text} opacity-70`}>
              {league.members} members • {league.isPublic ? 'Public' : 'Private'}
            </p>
          </div>
          <Crown className={`w-8 h-8 ${themeConfig.accent}`} />
        </div>
        
        <div className="space-y-3">
          <h4 className={`text-sm font-medium ${themeConfig.text}`}>Leaderboard</h4>
          {league.leaderboard.map((entry, index) => (
            <div key={index} className="flex items-center justify-between p-3 rounded-lg bg-white/10">
              <div className="flex items-center gap-3">
                <span className={`text-lg ${themeConfig.accent}`}>#{index + 1}</span>
                <span className={`text-2xl`}>{entry.user.avatar}</span>
                <span className={`${themeConfig.text}`}>{entry.user.name}</span>
              </div>
              <span className={`font-semibold ${themeConfig.accent}`}>
                {entry.steps.toLocaleString()}
              </span>
            </div>
          ))}
        </div>
      </motion.div>
    </div>
  );
};

// Battle Detail Screen Component
const BattleDetailScreen: React.FC<{ 
  theme: Theme; 
  battle: Battle; 
  user: User;
  onGoBack: () => void;
}> = ({ theme, battle, user, onGoBack }) => {
  const themeConfig = themes[theme];
  
  // Fiery theme override for battles
  const battleTheme = {
    ...themeConfig,
    cardBg: theme === 'black-neon' 
      ? 'bg-gradient-to-br from-red-900/80 to-orange-900/80 backdrop-blur-sm border border-red-500/30 shadow-lg shadow-red-500/20'
      : theme === 'glassmorphism'
      ? 'bg-gradient-to-br from-red-500/20 to-orange-500/20 backdrop-blur-xl border border-red-300/30 shadow-2xl shadow-black/20'
      : 'bg-gradient-to-br from-red-100 to-orange-100 backdrop-blur-sm border border-red-200 shadow-lg shadow-red-500/10',
    accent: theme === 'light' ? 'text-red-600' : 'text-red-400',
    glow: 'shadow-[0_0_20px_rgba(239,68,68,0.3)]'
  };

  return (
    <div className="space-y-6 pb-24">
      {/* Header */}
      <div className="flex items-center gap-4">
        <Button 
          onClick={onGoBack}
          variant="ghost"
          size="sm"
          className={themeConfig.text}
        >
          ← Back
        </Button>
        <div className="flex items-center gap-2">
          <Flame className={`w-6 h-6 ${battleTheme.accent}`} />
          <h1 className={`text-2xl font-bold ${themeConfig.text}`}>Battle Details</h1>
        </div>
      </div>

      {/* Battle Info */}
      <motion.div
        initial={{ opacity: 0, y: 20 }}
        animate={{ opacity: 1, y: 0 }}
        className={`${battleTheme.cardBg} rounded-3xl p-6 ${battleTheme.glow}`}
      >
        <div className="flex items-center justify-between mb-4">
          <div className="flex items-center gap-3">
            <span className="text-2xl">👤</span>
            <div>
              <h3 className={`font-semibold ${themeConfig.text}`}>{battle.opponent.name}</h3>
              <p className={`text-sm ${themeConfig.text} opacity-70`}>1v1 Battle</p>
            </div>
          </div>
          <div className="text-right">
            <p className={`text-sm ${battleTheme.accent} font-medium`}>Time Left</p>
            <p className={`text-lg font-bold ${battleTheme.accent}`}>{battle.timeLeft}</p>
          </div>
        </div>

        <div className="space-y-4">
          <div className="flex justify-between items-center">
            <span className={themeConfig.text}>You</span>
            <span className={`font-bold ${battleTheme.accent}`}>
              {battle.mySteps.toLocaleString()} steps
            </span>
          </div>
          <div className="flex justify-between items-center">
            <span className={themeConfig.text}>{battle.opponent.name}</span>
            <span className={`font-bold ${themeConfig.text} opacity-70`}>
              {battle.opponentSteps.toLocaleString()} steps
            </span>
          </div>
          
          {/* Battle progress bar */}
          <div className="relative h-4 bg-gray-700 rounded-full overflow-hidden">
            <motion.div
              className="absolute left-0 top-0 h-full bg-gradient-to-r from-red-500 to-orange-500"
              initial={{ width: 0 }}
              animate={{ 
                width: `${(battle.mySteps / (battle.mySteps + battle.opponentSteps)) * 100}%` 
              }}
              transition={{ duration: 1.5, ease: "easeOut" }}
            />
          </div>
          
          <div className="flex items-center justify-center gap-2">
            {battle.mySteps > battle.opponentSteps ? (
              <>
                <Zap className={`w-4 h-4 ${battleTheme.accent}`} />
                <span className={`text-sm font-medium ${battleTheme.accent}`}>
                  You're winning!
                </span>
              </>
            ) : (
              <span className={`text-sm ${themeConfig.text} opacity-70`}>
                Keep pushing!
              </span>
            )}
          </div>
        </div>
      </motion.div>
    </div>
  );
};

// Challenge Detail Screen Component
const ChallengeDetailScreen: React.FC<{ 
  theme: Theme; 
  challenge: Challenge; 
  onGoBack: () => void;
  onUpdateProgress: (challengeId: string, progress: number) => void;
}> = ({ theme, challenge, onGoBack, onUpdateProgress }) => {
  const themeConfig = themes[theme];

  return (
    <div className="space-y-6 pb-24">
      {/* Header */}
      <div className="flex items-center gap-4">
        <Button 
          onClick={onGoBack}
          variant="ghost"
          size="sm"
          className={themeConfig.text}
        >
          ← Back
        </Button>
        <h1 className={`text-2xl font-bold ${themeConfig.text}`}>Challenge Details</h1>
      </div>

      {/* Challenge Info */}
      <motion.div
        initial={{ opacity: 0, y: 20 }}
        animate={{ opacity: 1, y: 0 }}
        className={`${themeConfig.cardBg} rounded-3xl p-6`}
      >
        <div className="flex justify-between items-start mb-4">
          <div>
            <h3 className={`text-lg font-semibold ${themeConfig.text}`}>{challenge.title}</h3>
            <p className={`text-sm ${themeConfig.text} opacity-70`}>
              Due: {new Date(challenge.deadline).toLocaleDateString()}
            </p>
          </div>
          <Badge variant="outline" className={themeConfig.accent}>
            {challenge.type}
          </Badge>
        </div>
        
        <div className="space-y-4">
          <div className="flex justify-between">
            <span className={`text-sm ${themeConfig.text}`}>Progress</span>
            <span className={`text-sm ${themeConfig.accent}`}>
              {challenge.current} / {challenge.target}
            </span>
          </div>
          <div className="w-full bg-gray-700 rounded-full h-3">
            <motion.div
              className={`h-3 rounded-full ${
                theme === 'black-neon' ? 'bg-gradient-to-r from-cyan-400 to-blue-500' :
                theme === 'glassmorphism' ? 'bg-gradient-to-r from-purple-400 to-pink-400' :
                'bg-gradient-to-r from-purple-500 to-pink-500'
              }`}
              initial={{ width: 0 }}
              animate={{ width: `${(challenge.current / challenge.target) * 100}%` }}
              transition={{ duration: 1.5, ease: "easeOut" }}
            />
          </div>
          
          {/* Progress Update Buttons */}
          <div className="grid grid-cols-3 gap-2">
            <Button 
              onClick={() => onUpdateProgress(challenge.id, 100)}
              variant="outline"
              className="flex-1"
            >
              +100
            </Button>
            <Button 
              onClick={() => onUpdateProgress(challenge.id, 500)}
              variant="outline"
              className="flex-1"
            >
              +500
            </Button>
            <Button 
              onClick={() => onUpdateProgress(challenge.id, 1000)}
              variant="outline"
              className="flex-1"
            >
              +1000
            </Button>
          </div>
        </div>
      </motion.div>
    </div>
  );
};

// Main App Component
const PedometerApp: React.FC = () => {
  const [activeTab, setActiveTab] = useState('home');
  const [theme, setTheme] = useState<Theme>('black-neon');
  const [user, setUser] = useState(mockUser);
  const [leagues, setLeagues] = useState(mockLeagues);
  const [battles, setBattles] = useState(mockBattles);
  const [challenges, setChallenges] = useState(mockChallenges);
  
  // Navigation states
  const [currentView, setCurrentView] = useState<'main' | 'league-detail' | 'battle-detail' | 'challenge-detail'>('main');
  const [selectedLeague, setSelectedLeague] = useState<League | null>(null);
  const [selectedBattle, setSelectedBattle] = useState<Battle | null>(null);
  const [selectedChallenge, setSelectedChallenge] = useState<Challenge | null>(null);
  
  // Modal states
  const [showCreateLeagueModal, setShowCreateLeagueModal] = useState(false);
  const [showJoinLeagueModal, setShowJoinLeagueModal] = useState(false);
  const [showCreateBattleModal, setShowCreateBattleModal] = useState(false);
  const [showCreateChallengeModal, setShowCreateChallengeModal] = useState(false);
  
  // Form states
  const [newLeagueName, setNewLeagueName] = useState('');
  const [newLeagueIsPublic, setNewLeagueIsPublic] = useState(true);
  const [newChallengeTitle, setNewChallengeTitle] = useState('');
  const [newChallengeType, setNewChallengeType] = useState<'steps' | 'calories' | 'distance' | 'time' | 'battles'>('steps');
  const [newChallengeTarget, setNewChallengeTarget] = useState(10000);
  const [newChallengeDeadline, setNewChallengeDeadline] = useState('');
  const [newBattleOpponent, setNewBattleOpponent] = useState('');
  const [newBattleDuration, setNewBattleDuration] = useState<'24h' | '7d'>('24h');
  
  // Search states
  const [leagueSearchQuery, setLeagueSearchQuery] = useState('');
  const [leagueSearchResults, setLeagueSearchResults] = useState<League[]>([]);
  const [leagueJoinCode, setLeagueJoinCode] = useState('');
  const [nearbyLeagues, setNearbyLeagues] = useState<League[]>([]);
  
  // Settings states
  const [notificationsEnabled, setNotificationsEnabled] = useState(true);
  const [autoSyncEnabled, setAutoSyncEnabled] = useState(true);
  const [privacyModeEnabled, setPrivacyModeEnabled] = useState(false);
  
  const themeConfig = themes[theme];

  // Navigation functions
  const goToLeagueDetail = (league: League) => {
    setSelectedLeague(league);
    setCurrentView('league-detail');
  };

  const goToBattleDetail = (battle: Battle) => {
    setSelectedBattle(battle);
    setCurrentView('battle-detail');
  };

  const goToChallengeDetail = (challenge: Challenge) => {
    setSelectedChallenge(challenge);
    setCurrentView('challenge-detail');
  };

  const goBack = () => {
    setCurrentView('main');
    setSelectedLeague(null);
    setSelectedBattle(null);
    setSelectedChallenge(null);
  };

  // Load data from localStorage on mount
  useEffect(() => {
    const savedTheme = localStorage.getItem('stepwar-theme') as Theme;
    if (savedTheme) setTheme(savedTheme);
    
    const savedUser = localStorage.getItem('stepwar-user');
    if (savedUser) setUser(JSON.parse(savedUser));
    
    const savedLeagues = localStorage.getItem('stepwar-leagues');
    if (savedLeagues) setLeagues(JSON.parse(savedLeagues));
    
    const savedBattles = localStorage.getItem('stepwar-battles');
    if (savedBattles) setBattles(JSON.parse(savedBattles));
    
    const savedChallenges = localStorage.getItem('stepwar-challenges');
    if (savedChallenges) setChallenges(JSON.parse(savedChallenges));
    
    const savedSettings = localStorage.getItem('stepwar-settings');
    if (savedSettings) {
      const settings = JSON.parse(savedSettings);
      setNotificationsEnabled(settings.notifications ?? true);
      setAutoSyncEnabled(settings.autoSync ?? true);
      setPrivacyModeEnabled(settings.privacyMode ?? false);
    }
  }, []);

  // Save data to localStorage when it changes
  useEffect(() => {
    localStorage.setItem('stepwar-theme', theme);
  }, [theme]);

  useEffect(() => {
    localStorage.setItem('stepwar-user', JSON.stringify(user));
  }, [user]);

  useEffect(() => {
    localStorage.setItem('stepwar-leagues', JSON.stringify(leagues));
  }, [leagues]);

  useEffect(() => {
    localStorage.setItem('stepwar-battles', JSON.stringify(battles));
  }, [battles]);

  useEffect(() => {
    localStorage.setItem('stepwar-challenges', JSON.stringify(challenges));
  }, [challenges]);

  useEffect(() => {
    localStorage.setItem('stepwar-settings', JSON.stringify({
      notifications: notificationsEnabled,
      autoSync: autoSyncEnabled,
      privacyMode: privacyModeEnabled
    }));
  }, [notificationsEnabled, autoSyncEnabled, privacyModeEnabled]);

  // Function to add steps (simulate step counting)
  const addSteps = (steps: number) => {
    setUser(prev => ({
      ...prev,
      todaySteps: prev.todaySteps + steps,
      totalSteps: prev.totalSteps + steps
    }));
  };

  // Function to create a new league
  const createLeague = () => {
    if (!newLeagueName.trim()) return;
    
    const newLeague: League = {
      id: Date.now().toString(),
      name: newLeagueName,
      members: 1,
      isPublic: newLeagueIsPublic,
      leaderboard: [{ user, steps: user.todaySteps }]
    };
    
    setLeagues(prev => [...prev, newLeague]);
    setNewLeagueName('');
    setNewLeagueIsPublic(true);
    setShowCreateLeagueModal(false);
  };

  // Function to search leagues
  const searchLeagues = (query: string) => {
    setLeagueSearchQuery(query);
    if (!query.trim()) {
      setLeagueSearchResults([]);
      return;
    }
    
    // Simulate search results
    const mockSearchResults: League[] = [
      {
        id: 'search1',
        name: `Running Club ${query}`,
        members: Math.floor(Math.random() * 100) + 10,
        isPublic: true,
        leaderboard: []
      },
      {
        id: 'search2',
        name: `${query} Warriors`,
        members: Math.floor(Math.random() * 50) + 5,
        isPublic: true,
        leaderboard: []
      }
    ];
    setLeagueSearchResults(mockSearchResults);
  };

  // Function to join league by code
  const joinLeagueByCode = () => {
    if (!leagueJoinCode.trim()) return;
    
    const newLeague: League = {
      id: Date.now().toString(),
      name: `League ${leagueJoinCode}`,
      members: Math.floor(Math.random() * 20) + 5,
      isPublic: true,
      leaderboard: []
    };
    
    setLeagues(prev => [...prev, newLeague]);
    setLeagueJoinCode('');
    setShowJoinLeagueModal(false);
  };

  // Function to find nearby leagues
  const findNearbyLeagues = () => {
    const mockNearby: League[] = [
      {
        id: 'nearby1',
        name: 'Downtown Runners',
        members: 45,
        isPublic: true,
        leaderboard: []
      },
      {
        id: 'nearby2',
        name: 'Park Avenue Warriors',
        members: 23,
        isPublic: true,
        leaderboard: []
      }
    ];
    setNearbyLeagues(mockNearby);
  };

  // Function to create a new challenge
  const createChallenge = () => {
    if (!newChallengeTitle.trim() || !newChallengeDeadline) return;
    
    const newChallenge: Challenge = {
      id: Date.now().toString(),
      title: newChallengeTitle,
      type: newChallengeType,
      target: newChallengeTarget,
      current: 0,
      deadline: newChallengeDeadline
    };
    
    setChallenges(prev => [...prev, newChallenge]);
    setNewChallengeTitle('');
    setNewChallengeType('steps');
    setNewChallengeTarget(10000);
    setNewChallengeDeadline('');
    setShowCreateChallengeModal(false);
  };

  // Function to create a new battle
  const createBattle = (opponentName: string, duration: '24h' | '7d') => {
    const newBattle: Battle = {
      id: Date.now().toString(),
      opponent: { ...mockUser, name: opponentName },
      mySteps: user.todaySteps,
      opponentSteps: Math.floor(Math.random() * 5000) + 3000,
      status: 'active',
      timeLeft: duration === '24h' ? '23h 59m' : '6d 23h 59m'
    };
    
    setBattles(prev => [...prev, newBattle]);
    setNewBattleOpponent('');
    setNewBattleDuration('24h');
    setShowCreateBattleModal(false);
  };

  // Function to update challenge progress
  const updateChallengeProgress = (challengeId: string, progress: number) => {
    setChallenges(prev => prev.map(challenge => 
      challenge.id === challengeId 
        ? { ...challenge, current: Math.min(challenge.current + progress, challenge.target) }
        : challenge
    ));
  };

  const renderScreen = () => {
    if (currentView === 'league-detail' && selectedLeague) {
      return <LeagueDetailScreen 
        theme={theme} 
        league={selectedLeague} 
        user={user}
        onGoBack={goBack}
      />;
    }
    
    if (currentView === 'battle-detail' && selectedBattle) {
      return <BattleDetailScreen 
        theme={theme} 
        battle={selectedBattle} 
        user={user}
        onGoBack={goBack}
      />;
    }
    
    if (currentView === 'challenge-detail' && selectedChallenge) {
      return <ChallengeDetailScreen 
        theme={theme} 
        challenge={selectedChallenge} 
        onGoBack={goBack}
        onUpdateProgress={updateChallengeProgress}
      />;
    }

    switch (activeTab) {
      case 'home':
        return <HomeScreen theme={theme} user={user} onAddSteps={addSteps} />;
      case 'leagues':
        return <LeaguesScreen 
          theme={theme} 
          leagues={leagues}
          user={user}
          showCreateModal={showCreateLeagueModal}
          showJoinModal={showJoinLeagueModal}
          onShowCreateModal={setShowCreateLeagueModal}
          onShowJoinModal={setShowJoinLeagueModal}
          newLeagueName={newLeagueName}
          newLeagueIsPublic={newLeagueIsPublic}
          onNewLeagueNameChange={setNewLeagueName}
          onNewLeagueIsPublicChange={setNewLeagueIsPublic}
          onCreateLeague={createLeague}
          searchQuery={leagueSearchQuery}
          searchResults={leagueSearchResults}
          onSearch={searchLeagues}
          joinCode={leagueJoinCode}
          onJoinCodeChange={setLeagueJoinCode}
          onJoinByCode={joinLeagueByCode}
          nearbyLeagues={nearbyLeagues}
          onFindNearby={findNearbyLeagues}
          onLeagueClick={goToLeagueDetail}
        />;
      case 'battles':
        return <BattlesScreen 
          theme={theme} 
          battles={battles}
          user={user}
          showCreateModal={showCreateBattleModal}
          onShowCreateModal={setShowCreateBattleModal}
          onCreateBattle={createBattle}
          newOpponent={newBattleOpponent}
          newDuration={newBattleDuration}
          onNewOpponentChange={setNewBattleOpponent}
          onNewDurationChange={setNewBattleDuration}
          onBattleClick={goToBattleDetail}
        />;
      case 'challenges':
        return <ChallengesScreen 
          theme={theme} 
          challenges={challenges}
          showCreateModal={showCreateChallengeModal}
          onShowCreateModal={setShowCreateChallengeModal}
          newChallengeTitle={newChallengeTitle}
          newChallengeType={newChallengeType}
          newChallengeTarget={newChallengeTarget}
          newChallengeDeadline={newChallengeDeadline}
          onNewChallengeTitleChange={setNewChallengeTitle}
          onNewChallengeTypeChange={setNewChallengeType}
          onNewChallengeTargetChange={setNewChallengeTarget}
          onNewChallengeDeadlineChange={setNewChallengeDeadline}
          onCreateChallenge={createChallenge}
          onUpdateProgress={updateChallengeProgress}
          onChallengeClick={goToChallengeDetail}
        />;
      case 'profile':
        return <ProfileScreen 
          theme={theme} 
          onThemeChange={setTheme} 
          user={user}
          notificationsEnabled={notificationsEnabled}
          autoSyncEnabled={autoSyncEnabled}
          privacyModeEnabled={privacyModeEnabled}
          onNotificationsChange={setNotificationsEnabled}
          onAutoSyncChange={setAutoSyncEnabled}
          onPrivacyModeChange={setPrivacyModeEnabled}
        />;
      default:
        return <HomeScreen theme={theme} user={user} onAddSteps={addSteps} />;
    }
  };

  return (
    <div className={`min-h-screen ${themeConfig.bg} transition-all duration-500`}>
      <div className="max-w-md mx-auto relative">
        {/* Status Bar */}
        <div className={`flex justify-between items-center p-4 ${themeConfig.text}`}>
          <span className="text-sm font-medium">{new Date().toLocaleTimeString('en-US', { hour: '2-digit', minute: '2-digit' })}</span>
          <div className="flex items-center gap-1">
            <div className="w-4 h-2 border border-current rounded-sm">
              <div className="w-3/4 h-full bg-current rounded-sm" />
            </div>
          </div>
        </div>

        {/* Main Content */}
        <div className="px-4">
          <AnimatePresence mode="wait">
            <motion.div
              key={activeTab}
              initial={{ opacity: 0, x: 20 }}
              animate={{ opacity: 1, x: 0 }}
              exit={{ opacity: 0, x: -20 }}
              transition={{ duration: 0.3 }}
            >
              {renderScreen()}
            </motion.div>
          </AnimatePresence>
        </div>

        {/* Bottom Navigation */}
        <BottomNav 
          activeTab={activeTab} 
          onTabChange={setActiveTab} 
          theme={theme}
        />
      </div>
    </div>
  );
};

export default function App() {
  return <PedometerApp />;
}
